### PR TITLE
perf: improve flashy loading while lazy loading

### PR DESCRIPTION
**Pull request by soknaly**
### Improve flashy loading when lazy loading

After introducing lazy loading, I noticed an unpleasant User Experience when the pages were loaded along with the sidebar. So instead of lazy loading the whole page, we only lazy load the sub-page which so the loading indicator only shows in the content panel at the right side of the sidebar.

Following `react-router-dom` v6, we can take advantage of `Outlet` to simplify this.

**This change includes the following**
- Breadcrumbs link will be no longer specified on each page, instead, we use the function to build breadcrumbs links for each path map.
- Breadcrumbs are now generated by using the function `BuildBreadcrumbLinksByTitles`.
  - Project -> Monitors -> View Monitor -> Custom Fields represented by `/dashboard/:projectId/monitors/:id/custom-fields`
  - Breaking down breadcrumb with a link representation for each item (reversed)
     - Custom Fields = `/dashboard/:projectId/minitors/:id/custom-fields` (current pathname)
     - View Monitor = `/dashboard/:projectId/monitors/:id` (current pathname - 1)
     - Monitors = `/dashboard/:projectId/monitors` (current pathname - 2)
     - Project = `/dashboard/:projectId` (current pathname - 3)
 - This is how I expected breadcrumbs to be. If in any case breadcrumbs behave differently, we can use this `BuildBreadcrumbLinks`
- Each sub-page no longer needs to use `ModelPage` as it's been covered in its container component along with `Outlet` component.

### Pull Request Checklist: 

- [ ] Please make sure all jobs pass before requesting a review. 
- [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).
- [x] Have you lint your code locally before submission?
- [ ] Did you write tests where appropriate?

#### Before

https://github.com/OneUptime/oneuptime/assets/20983608/02b6450c-6b32-4c2c-8fc2-d585b949a386

#### After

https://github.com/OneUptime/oneuptime/assets/20983608/44c3130f-b62a-41cf-a2ae-c16aed1682bd

### CODE REVIEW

**Comment by soknaly** [comment]
@simlarsen @nawazdhandala You can check this one if it makes sense, I can continue working on this for other pages.

If you have questions or concerns, please let me know.

Thank you

**Comment by nawazdhandala** [comment]
Yes, please do.

**Comment by simlarsen** [comment]
@soknaly  Thank you for all your work! Happy new year to you and family.

**Comment by soknaly** [comment]
Thank you @simlarsen! Happy New Year to you and your family too 🎆

File: CommonUI/src/Utils/Navigation.ts [hunk 1]
```diff
@@ -31,6 +37,21 @@ abstract class Navigation {
         return new Route(window.location.pathname);
     }
 
+    public static getBreadcrumpRoute(level: number): Route {
```
Comment (simlarsen) [inline CommonUI/src/Utils/Navigation.ts:40]: typo

**Review by simlarsen — CHANGES_REQUESTED** [review]

File: CommonUI/src/Config.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ export const HTTP_PROTOCOL: Protocol =
 
 export const HOST: string = env('HOST') || '';
 
-export const BILLING_ENABLED: boolean = env('BILLING_ENABLED') === 'true';
+export const BILLING_ENABLED: boolean = true;
```
[the remark this answers is not in the document]
Comment (simlarsen) [reply CommonUI/src/Config.ts:39]: or I can fix it. Not a big deal.

File: CommonUI/src/Utils/Navigation.ts (hunk 1 above)
Comment (simlarsen) [reply CommonUI/src/Utils/Navigation.ts:40]: or I can fix it. Not a big deal.

File: CommonUI/src/Config.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (soknaly) [reply CommonUI/src/Config.ts:39]: @simlarsen Sorry forgot to rever this.

**Pull request merged**